frontend/packages/topology/src/behavior/usePanZoom.tsx

```diff
@@ -5,134 +5,118 @@ import { action, autorun, IReactionDisposer } from 'mobx';
 import ElementContext from '../utils/ElementContext';
 import useCallbackRef from '../utils/useCallbackRef';
 import Point from '../geom/Point';
-import { isGraph, ModelKind } from '../types';
+import { Graph, isGraph, ModelKind } from '../types';
 import { ATTR_DATA_KIND } from '../const';
 
-export type PanZoomTransform = {
-  x: number;
-  y: number;
-  k: number;
-};
-
-const ZOOM_EXTENT: [number, number] = [0.25, 4];
-
 export type PanZoomRef = (node: SVGGElement | null) => void;
 
 // Used to send events prevented by d3.zoom to the document allowing modals, dropdowns, etc, to close
 const propagatePanZoomMouseEvent = (e: Event): void => {
   document.dispatchEvent(new MouseEvent(e.type, e));
 };
 
-export const usePanZoom = (zoomExtent: [number, number] = ZOOM_EXTENT): PanZoomRef => {
+export const usePanZoom = (): PanZoomRef => {
   const element = React.useContext(ElementContext);
   if (!isGraph(element)) {
     throw new Error('usePanZoom must be used within the scope of a Graph');
   }
   const elementRef = React.useRef(element);
-  elementRef.current = element;
+  elementRef.current = element as Graph;
 
-  const refCallback = useCallbackRef<PanZoomRef>(
-    React.useCallback(
-      (node: SVGGElement | null) => {
-        let disposeListener: IReactionDisposer | undefined;
-        if (node) {
-          // TODO fix any type
-          const $svg = d3.select(node.ownerSVGElement) as any;
-          if (node && node.ownerSVGElement) {
-            node.ownerSVGElement.addEventListener('mousedown', propagatePanZoomMouseEvent);
-            node.ownerSVGElement.addEventListener('click', propagatePanZoomMouseEvent);
+  return useCallbackRef<PanZoomRef>((node: SVGGElement | null) => {
+    let disposeListener: IReactionDisposer | undefined;
+    if (node) {
+      // TODO fix any type
+      const $svg = d3.select(node.ownerSVGElement) as any;
+      if (node && node.ownerSVGElement) {
+        node.ownerSVGElement.addEventListener('mousedown', propagatePanZoomMouseEvent);
+        node.ownerSVGElement.addEventListener('click', propagatePanZoomMouseEvent);
+      }
+      const zoom = d3
+        .zoom()
+        .scaleExtent([elementRef.current.getMinScale(), elementRef.current.getMaxScale()])
```

Comment:
This behavior needs to track changes to the min and max scales and ensure that the new extent is set when one of these values change.